### PR TITLE
Improve Accuracy of Streaming Triangle Counting

### PR DESCRIPTION
**Pull request by Ashok12011234**

### CODE REVIEW

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 1]
```diff
@@ -1464,9 +1469,24 @@ static void triangles_command(std::string masterIP, int connFd, SQLiteDBInterfac
     }
 }
 
+void JasmineGraphFrontEnd::scheduleStrianJobs(JobRequest &jobDetails, std::priority_queue<JobRequest> &jobQueue,
+                                               JobScheduler *jobScheduler, bool *strian_exist) {
+    while (true) {
```
Comment (miyurud) [inline src/frontend/JasmineGraphFrontEnd.cpp:1474]: Lets avoid use of `while(true) {}`.

**Review by miyurud — CHANGES_REQUESTED** [review]
I have put few comments. Please address all of them.

File: src/frontend/JasmineGraphFrontEnd.cpp (hunk 1 above)
Comment (Ashok12011234) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Done

Comment (miyurud) [reply src/frontend/JasmineGraphFrontEnd.cpp]: Ok

File: src/query/algorithms/triangles/StreamingTriangles.cpp [hunk 2]
```diff
@@ -159,40 +156,44 @@ NativeStoreTriangleResult StreamingTriangles::countDynamicLocalTriangles(
...
+        long targetNode = std::stol(relationBlock->getDestination()->id);
+        edges.push_back(std::make_pair(sourceNode, targetNode));
+        edges.push_back(std::make_pair(targetNode, sourceNode));
         newAdjacencyList[sourceNode].insert(targetNode);
         newAdjacencyList[targetNode].insert(sourceNode);
+        localAdjacenyList[sourceNode].insert(targetNode);
+        localAdjacenyList[targetNode].insert(sourceNode);
     }
 
+    std::map<long, std::unordered_set<long>> adjacenyList = localAdjacenyList;
```
Comment (miyurud) [inline src/query/algorithms/triangles/StreamingTriangles.cpp:193]:
```suggestion
    std::map<long, std::unordered_set<long>> adjacencyList = localAdjacencyList;
```

File: src/query/algorithms/triangles/StreamingTriangles.cpp [hunk 3]
```diff
@@ -159,40 +156,44 @@ NativeStoreTriangleResult StreamingTriangles::countDynamicLocalTriangles(
...
-
-        edges.push_back(std::make_pair(std::stol(relationBlock->getDestination()->id),
-                                           std::stol(relationBlock->getSource()->id)));
+        long sourceNode = std::stol(relationBlock->getSource()->id);
+        long targetNode = std::stol(relationBlock->getDestination()->id);
+        edges.push_back(std::make_pair(sourceNode, targetNode));
+        edges.push_back(std::make_pair(targetNode, sourceNode));
+        newAdjacencyList[sourceNode].insert(targetNode);
+        newAdjacencyList[targetNode].insert(sourceNode);
+        localAdjacenyList[sourceNode].insert(targetNode);
```
Comment (miyurud) [inline src/query/algorithms/triangles/StreamingTriangles.cpp:177]:
```suggestion
        localAdjacencyList[sourceNode].insert(targetNode);
```

File: src/query/algorithms/triangles/StreamingTriangles.cpp [hunk 4]
```diff
@@ -206,6 +207,7 @@ std::string StreamingTriangles::countDynamicCentralTriangles(
         std::vector<std::string> oldCentralRelationCount) {
     streaming_triangle_logger.info("###STREAMING TRIANGLE### Dynamic Streaming Central Triangle "
                                   "Counting: Started");
+    std::string joinedString = partitionIdList[0] + partitionIdList[1] + partitionIdList[2];
```
Comment (miyurud) [inline src/query/algorithms/triangles/StreamingTriangles.cpp:210]: Are we sure that always the `partitionIdList` size is not less than 3?

File: src/query/algorithms/triangles/StreamingTriangles.h [hunk 5]
```diff
@@ -35,6 +35,8 @@ struct NativeStoreTriangleResult {
 
 class StreamingTriangles {
  public:
+    static std::map<long, std::unordered_set<long>> localAdjacenyList;
```
Comment (miyurud) [inline src/query/algorithms/triangles/StreamingTriangles.h:38]:
```suggestion
    static std::map<long, std::unordered_set<long>> localAdjacencyList;
```

**Review by miyurud — CHANGES_REQUESTED** [review]
I have put some more comments. Please address those.

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 2 above)
Comment (Ashok12011234) [reply src/query/algorithms/triangles/StreamingTriangles.cpp:193]: DONE

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 3 above)
Comment (Ashok12011234) [reply src/query/algorithms/triangles/StreamingTriangles.cpp:177]: Done

File: src/query/algorithms/triangles/StreamingTriangles.h (hunk 5 above)
Comment (Ashok12011234) [reply src/query/algorithms/triangles/StreamingTriangles.h:38]: Done

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 4 above)
Comment (Ashok12011234) [reply src/query/algorithms/triangles/StreamingTriangles.cpp]: I have solved this

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 3 above)
Comment (miyurud) [reply src/query/algorithms/triangles/StreamingTriangles.cpp]: Ok

File: src/query/algorithms/triangles/StreamingTriangles.h (hunk 5 above)
Comment (miyurud) [reply src/query/algorithms/triangles/StreamingTriangles.h]: Ok

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 4 above)
Comment (miyurud) [reply src/query/algorithms/triangles/StreamingTriangles.cpp]: Ok

File: src/query/algorithms/triangles/StreamingTriangles.cpp (hunk 2 above)
Comment (miyurud) [reply src/query/algorithms/triangles/StreamingTriangles.cpp]: Ok

**Review by miyurud — APPROVED** [review]

**Pull request merged**